src/sign.rs

```diff
@@ -650,514 +804,318 @@ mod tests {
+        fn cose_sign1_ec256_text_tagged() {
+            let (ec_private, ec_public) = generate_ec256_test_key();
+            let mut map = HeaderMap::new();
+            map.insert(CborValue::Integer(4), CborValue::Bytes(b"11".to_vec()));
+
+            let cose_doc1 = CoseSign1::new(TEXT, &map, &ec_private).unwrap();
+            let tagged_bytes = cose_doc1.as_bytes(true).unwrap();
+            // Tag 6.18 should be present
+            assert_eq!(tagged_bytes[0], 6 << 5 | 18);
+            // The value should be a sequence
+            assert_eq!(tagged_bytes[1], 4 << 5 | 4);
+            let cose_doc2 = CoseSign1::from_bytes(&tagged_bytes).unwrap();
+
+            assert_eq!(
+                cose_doc1.get_payload(None).unwrap(),
+                cose_doc2.get_payload(Some(&ec_public)).unwrap()
+            );
+        }
 
-    #[test]
-    fn cose_sign1_ec256_text_tagged() {
-        let (ec_private, ec_public) = generate_ec256_test_key();
-        let mut map = HeaderMap::new();
-        map.insert(CborValue::Integer(4), CborValue::Bytes(b"11".to_vec()));
-
-        let cose_doc1 = CoseSign1::new(TEXT, &map, &ec_private).unwrap();
-        let tagged_bytes = cose_doc1.as_bytes(true).unwrap();
-        // Tag 6.18 should be present
-        assert_eq!(tagged_bytes[0], 6 << 5 | 18);
-        // The value should be a sequence
-        assert_eq!(tagged_bytes[1], 4 << 5 | 4);
-        let cose_doc2 = CoseSign1::from_bytes(&tagged_bytes).unwrap();
-
-        assert_eq!(
-            cose_doc1.get_payload(None).unwrap(),
-            cose_doc2.get_payload(Some(&ec_public)).unwrap()
-        );
-    }
+        #[test]
+        fn cose_sign1_ec256_text_with_extra_protected() {
+            let (ec_private, ec_public) = generate_ec256_test_key();
+
+            let mut protected = HeaderMap::new();
+            protected.insert(
+                CborValue::Integer(1),
+                (SignatureAlgorithm::ES256 as i8).into(),
+            );
+            protected.insert(CborValue::Integer(15), CborValue::Bytes(b"12".to_vec()));
+
+            let mut unprotected = HeaderMap::new();
+            unprotected.insert(CborValue::Integer(4), CborValue::Bytes(b"11".to_vec()));
+
+            let cose_doc1 =
+                CoseSign1::new_with_protected(TEXT, &protected, &unprotected, &ec_private).unwrap();
+            let cose_doc2 = CoseSign1::from_bytes(&cose_doc1.as_bytes(false).unwrap()).unwrap();
+
+            let (protected, payload) = cose_doc2
+                .get_protected_and_payload(Some(&ec_public))
+                .unwrap();
+
+            assert_eq!(
+                protected.get(&CborValue::Integer(1)),
+                Some(&CborValue::Integer(-7)),
+            );
+            assert_eq!(
+                protected.get(&CborValue::Integer(15)),
+                Some(&CborValue::Bytes(b"12".to_vec())),
+            );
+            assert_eq!(payload, TEXT,);
+        }
 
-    #[test]
-    fn cose_sign1_ec256_text_tagged_serde() {
```

Comment:
This test seems to be missing, but I can't quite figure out why.

Reply:
Because I accidentally dropped that when resolving the merge conflicts.